delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/jsonld/context/RemoteContextResolution.scala

```diff
@@ -0,0 +1,54 @@
+package ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context
+
+import akka.http.scaladsl.model.Uri
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.rdf.implicits._
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.RemoteContextResolutionError.RemoteContextCircularDependency
+import io.circe.Json
+import monix.bio.IO
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.RemoteContextResolution.Result
+
+trait RemoteContextResolution {
+
+  /**
+    * Resolve a passed ''uri''.
+    *
+   * @return the expected Json payload response from the passed ''uri''
+    */
+  def resolve(uri: Uri): Result[Json]
```

Comment:
Context resolution happens over the wire, over HTTP, which does not support IRIs. 
In the JSON-LD spec the meontions to remote context are saying it is a Uri and also in Json-LD Java implementation and titanium

Reply:
But we don't allow that and instead we use resource identifiers which can be iris, right? We could transform the resource ids into Uris and back during the resolution, but does it make sense?

Reply:
Ok. It seems to work with IRIs as well as with Uris. I changed the signature to IRI since it is true that the Nexus resources can be IRIs, not just Uris